packages/aws-cdk-lib/aws-cloudfront/lib/function.ts

```diff
@@ -222,11 +222,13 @@ export class Function extends Resource implements IFunction {
   }
 
   private generateName(): string {
-    const name = Stack.of(this).region + Names.uniqueId(this);
-    if (name.length > 64) {
-      return name.substring(0, 32) + name.substring(name.length - 32);
+    const maxNameLength = 64 - '${Token[AWS.Region.00]}'.length;
```

Comment:
nit: this variable is not really the max name length
nit2: lets add a doc string here to explain what & why we are doing this. It is not clear since this was an edge case we are trying to avoid.